frontend/coin-control: add address reuse warning

strmci opened the pull request:
Add a warning to the coin control modal when a user selects one or more UTXOs from reused addresses. The warning: "One or more UTXOs have a reused address. Be aware  the receiver will see all UTXOs associated with those addresses."

The UTXOs now have red badge when their address was re-used.

Beerosagos reviewed (changes requested):
Nice! Just noticed a small bug.

I also feel like the duplicated address should be somehow highlighted. Otherwise it can be hard to find those out.

strmci commented:
@Beerosagos @thisconnect 

- fixed the repeated warning for each UTXO's script type
- the warning is shown only when the user selects the UTXO from the reused address
- I changed the warning from `Status` to `Message` so it is not all red
- added a badge to all UTXOs from re-used addresses
- CHANGELOG updated

![image](https://github.com/BitBoxSwiss/bitbox-wallet-app/assets/36126939/a736458b-64be-4e6f-bc21-b29f43a461ee)

PTAL

Beerosagos commented on frontends/web/src/routes/account/send/utxos.tsx:
```diff
@@ -56,6 +58,7 @@ export const UTXOs = ({
   const { t } = useTranslation();
   const [utxos, setUtxos] = useState<TUTXO[]>([]);
   const [selectedUTXOs, setSelectedUTXOs] = useState<TSelectedUTXOs>({});
+  const [showAddressReusedMessage, setShowAddressReusedMessage] = useState(0);
```
nit: I find the name of this const slightly misleading. Maybe smth like `selectedReusedAddresses`? Since it is a number and not a bool.

strmci replied:
Updated to `reusedAddressUTXOs`.

Beerosagos reviewed (approved):
LGTM with a couple nits :+1: 

Nit: maybe we can vertically center better the badge?

![image](https://github.com/BitBoxSwiss/bitbox-wallet-app/assets/11994448/33018a08-69fd-4930-a35c-f5b791ca02f2)

thisconnect commented:
> Nit: maybe we can vertically center better the badge?

I'll take a look, ideally badge would be nice and inline with text. Not sure yet what causes this.

thisconnect commented:
> I'll take a look, ideally badge would be nice and inline with text. Not sure yet what causes this.

tl;dr it was a combination of inconsitent line-heights (parent element) and vertical centering. 

branch with fix
https://github.com/thisconnect/bitbox-wallet-app/tree/frontend-fix-badge-centering

diff
https://github.com/BitBoxSwiss/bitbox-wallet-app/compare/master...thisconnect:frontend-fix-badge-centering?expand=1

@strmci @Beerosagos let me know if you want to merge this PR first and if I should open a followup PR or feel free to chery-pick my commit into this change.

strmci commented:
@thisconnect
I cherry-picked your Badge improvement, thanks.
